src/loaders/utils/GetTakeoverDetails.ts

```diff
@@ -0,0 +1,57 @@
+import { isServerSide } from 'utils/getEnv';
+import { getArticleVideoServiceBaseUrl } from 'utils/getPandoraDetails';
+import { trackYoutubeTakeoverDetails, YoutubePlayerTrackingProps } from 'youtube/players/shared/youtubeTrackingEvents';
+import { getMediaWikiConfigDetails } from 'loaders/utils/GetMediaWikiConfigDetails';
+import { TakeoverDetails, TakeoverResponse } from 'loaders/types';
+
+function getTakeoverUrl(wikiId: string): string {
+	const articleVideoBaseUrl = getArticleVideoServiceBaseUrl();
+	if (articleVideoBaseUrl.length === 0) {
+		return articleVideoBaseUrl;
+	}
+	if (!wikiId) {
+		return '';
+	}
+	return `${articleVideoBaseUrl}takeover/v1/takeover-mappings/${wikiId}`;
+}
+
+export async function getTakeoverDetails({ deviceType }: YoutubePlayerTrackingProps) {
+	const { wikiId, isTier3Wiki } = getMediaWikiConfigDetails();
+	if (!wikiId || isTier3Wiki) {
+		return null;
+	}
+
+	const response: Response = await fetch(getTakeoverUrl(wikiId));
+	const dataArray: TakeoverResponse[] = await response.json();
+	const takeoverDetails: TakeoverDetails = { videoId: null, type: null };
```

Comment:
I think it's better to return undefined or null if there is no takeover.

So we could have:

```
if(!data) {
  return;
}

return {
  type: data.youtube_take_over ? 'youtube' : 'vimeo',
  videoId: data.takeover_video_id
}

Reply:
Done